File: Arcade/Intro/Java/CommonCharacterCount.java
import java.util.stream.IntStream;
import java.util.stream.Collectors;

final class CommonCharacterCount {
	
	int commonCharacterCount(String s1, String s2) {
	    int o = 0;
	    String common = IntStream
	        .range(0x61, 0x7b)
	        .filter(n -> s1.contains((char)n+"") && s2.contains((char)n+""))
	        .boxed().map(c -> (char)c.intValue()+"").collect(Collectors.joining());
	    
	    for(char c : common.toCharArray()) {
	        o += Math.min(
	            s1.chars().map(n -> c == (char)n ? 1 : 0).sum(),
	            s2.chars().map(n -> c == (char)n ? 1 : 0).sum());
	    }
	    
	    return o;
	}
}

File: Arcade/Intro/javascript/commonCharacterCount.js
function commonCharacterCount(s1, s2) {
    let common = [], o = 0;
    for(let i = 0x61; i < 0x7b; i++) {
        let s = String.fromCharCode(i);
        if(s1.includes(s) && s2.includes(s)) common.push(s);
    }
    
    for(let e of common) {
        let p = 0, q = 0;
        for(let v of s1) if(v == e) p++;
        for(let v of s2) if(v == e) q++;
        o += Math.min(p, q);
    }
    
    return o;
}